katdal/datasources.py

```diff
@@ -152,25 +154,27 @@ def __init__(self, telstate, capture_block_id=None, stream_name=None,
                 sensors[key] = TelstateSensorData(telstate, key)
         metadata = AttrsSensors(telstate, sensors, name=source_name)
         try:
-            base_name = telstate['chunk_name']
             chunk_info = telstate['chunk_info']
         except KeyError:
             # Metadata without data
             DataSource.__init__(self, metadata, None)
         else:
-            # Extract VisFlagsWeights and timestamps from telstate
-            with tempfile.NamedTemporaryFile() as f:
-                f.write(telstate['ceph_conf'])
-                f.flush()
-                pool = telstate['ceph_pool']
-                store = RadosChunkStore.from_config(f.name, pool)
-            ts_name = store.join(base_name, 'timestamps')
+            try:
+                store = S3ChunkStore.from_url(telstate['s3_endpoint_url'])
```

Comment:
You'll get a KeyError if `s3_endpoint_url` isn't there.

Reply:
Fixed. S3 is also now the only option.